src/main/java/com/conveyal/r5/profile/FastRaptorWorker.java

```diff
@@ -546,33 +546,73 @@ private void doScheduledSearchForRound (RaptorState outputState) {
                             }
                         }
                     } else {
-                        // We are already on a trip, but check if this stop was reached early enough to back up
-                        // to an earlier trip on the same pattern.
-                        // For example, consider two potential boarding stops on a pattern (A and B), with the search
-                        // origin closer to B:
-                        //  A----B
-                        //       |
-                        //       O
-                        // In this setup, the earliestBoardTime at A is less than (before) the earliestBoardTime at B.
-                        // It might be possible to board a certain trip at B, but only a later trip at A. Because A
-                        // is evaluated first (providing updates to downstream stops), we need to check for earlier
-                        // trips that could be boarded at B.
-                        int bestTripIdx = onTrip;
-                        while (--bestTripIdx >= 0) {
-                            TripSchedule trip = pattern.tripSchedules.get(bestTripIdx);
-                            if (trip.headwaySeconds != null || !servicesActive.get(trip.serviceCode)) {
+
+                        // We have already boarded a trip on this pattern at an upstream stop; check for cases in which
+                        // it would be preferable to board the pattern at this stop instead.
+
+                        TripSchedule onTripSchedule = pattern.tripSchedules.get(onTrip);
+
+                        // First, check whether boarding the same trip at this stop instead of the upstream one would
+                        // be preferred. This will not affect total travel time, but it will affect the breakdown of
+                        // walk vs. wait. So if needed as an optimization, these blocks could be disabled unless
+                        // request.computePaths = true;
+                        boolean boardHereInstead = false;
```

Comment:
We should factor this whole decision whether to board here out into a method, and make that method call and the subsequent `if (boardHereInstead)` clause all conditional on request.computePaths.